packages/ra-core/src/form/sanitizeEmptyValues.ts

```diff
@@ -0,0 +1,33 @@
+import merge from 'lodash/merge';
+
+/**
+ * Because final-form removes undefined and empty string values completely
+ * (the key for the empty field is removed from the values), we have to check
+ * wether this value was initially provided so that it is correctly sent to
+ * the backend.
+ * See https://github.com/final-form/react-final-form/issues/130#issuecomment-493447888
+ *
+ * @param initialValues The initial values provided to the form
+ * @param values The current form values
+ */
+const sanitizeEmptyValues = (initialValues: object, values: object) => {
+    // For every field initialy provided, we check wether it value has been removed
+    // and set it explicitly to an empty string
+    const initialValuesWithEmptyFields = Object.keys(initialValues).reduce(
+        (acc, key) => {
+            if (typeof values[key] === 'object' && values[key] !== null) {
+                acc[key] = sanitizeEmptyValues(initialValues[key], values[key]);
+            } else {
+                acc[key] =
+                    typeof values[key] === 'undefined' ? '' : values[key];
+            }
```

Comment:
As far as I understand you consider 3 cases:
- values[key] is undefined, in that case you initialize the value with ""
- values[key] is a non-null object, and you recursively sanitarize the values
- in all other cases, the value is kept as it is
I think you should also test that initialValues[key]!==null otherwise a recursive call may occurs and throw an error (Object.keys(null) is not defined)
Furthermore, if value[key]===nulll you keep the null value, is that the wanted behavior ?